### PR TITLE
[libra framework] Check that account is DD in publish_preburn.

### PR DESCRIPTION
**Pull request by tzakian**
Also uncomments an assert and makes configs only publishable at genesis.

### CODE REVIEW

File: language/stdlib/modules/DesignatedDealer.move [hunk 1]
```diff
@@ -114,7 +114,7 @@ module DesignatedDealer {
         assert(Roles::has_treasury_compliance_role(tc_account), EACCOUNT_NOT_TREASURY_COMPLIANCE);
         let tiers = &mut borrow_global_mut<TierInfo<CoinType>>(dd_addr).tiers;
         let number_of_tiers = Vector::length(tiers);
-        assert(number_of_tiers + 1 <= MAX_NUM_TIERS, EINVALID_TIER_ADDITION);
+        assert(number_of_tiers <= MAX_NUM_TIERS, EINVALID_TIER_ADDITION);
```
Comment (shazqadeer) [inline language/stdlib/modules/DesignatedDealer.move:117]: This change does not look correct to me.  tiers has not been updated yet.  We have to make sure once a tier is added, the number of tiers continues to be bounded by MAX_NUM_TIERS.

**Comment by junkil-park** [comment]
> [move-prover] Move Prover test (`language/move-prover/cargo test`) on commit [d3e3817](https://github.com/libra/libra/commit/d3e381779572ac4d55a5b5b43e215b5baa68a609) did **fail**. See [details](https://github.com/mimoo/mirai-bot/runs/875017326?check_suite_focus=true#step:7:1).

A timeout occurred in the verification of `TransactionFee:: preburn_burn_fees`. I don't see any changes in this PR that can break the spec of the function. Then, could the verification for that function be turned off, @emmazzz?

**Comment by tzakian** [comment]
I'll turn off that check for now and add a todo comment for it.

**Comment by junkil-park** [comment]
> I'll turn off that check for now and add a todo comment for it.

I was writing this: 
@tzakian, @emmazzz said that you could turn off the verification for the functions causing timeouts ("pragma verify = false;"). She would take a look at them once it's landed.

**Comment by tzakian** [comment]
Tried things out locally with T=20 and it all passed. I'm going to start the land if if mirai-bot complains again I'll stop it.

@bors-libra r=junkil-park

**Pull request closed without merging**